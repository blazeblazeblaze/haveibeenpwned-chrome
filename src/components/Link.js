import React from 'react';
import PropTypes from 'prop-types';

const Link = ({onClick, className, href, children}) => {
  return (
    <a href={href} onClick={onClick} className={className}>
      {children}
    </a>
  );
}

Link.propTypes = {
  onClick: PropTypes.func,
  className: PropTypes.string,
  href: PropTypes.string
};

Link.defaultProps = {
  onClick: () => {},
  className: "",
  href: ""
};

export default Link;
