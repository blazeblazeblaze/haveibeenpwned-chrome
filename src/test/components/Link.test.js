import React from 'react';
import ReactDOM from 'react-dom';
import Link from '../../components/Link';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Link />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Link renders with no attrs', () => {
  const component = renderer.create(
    <Link />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Link renders with classes and href', () => {
  const component = renderer.create(
    <Link className='custom-class' href='http://google.com' onClick={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Link renders with its children', () => {
  const component = renderer.create(
    <Link>
      <span>hello world</span>
    </Link>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
