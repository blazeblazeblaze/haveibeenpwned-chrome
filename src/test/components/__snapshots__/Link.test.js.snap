// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders with classes and href 1`] = `
<a
  className="custom-class"
  href="http://google.com"
  onClick={[Function]}
/>
`;

exports[`Link renders with its children 1`] = `
<a
  className=""
  href=""
  onClick={[Function]}
>
  <span>
    hello world
  </span>
</a>
`;

exports[`Link renders with no attrs 1`] = `
<a
  className=""
  href=""
  onClick={[Function]}
/>
`;
